apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -44,6 +44,11 @@ public static object ConvertToClrType(this Value wireValue, V1.Type spannerType,
                 }
                 return null;
             }
+            if (targetClrType == typeof(object))
```

Comment:
Not exactly sure what you mean.  if someone asks for the value of a cell with "typeof(object)", they are saying they dont care about the type -- so we just give the 'expected type' which we calculate based on the spanner type on the wire.
There were some changes I had to make around this due to the fact that DataAdapter does not support nullable types.  So the default conversions had to be non-nullable -- but if you ask for typeof(object), you would expect 'null' back instead of default(T) (and in fact DataAdapter expects that behavior).  So some delicate type handling.  All of this needs quite a bit of unit test coverage.